File: src/java/org/voegtle/weatherstation/server/request/DataType.java
package org.voegtle.weatherstation.server.request;

public enum DataType {
  AGGREGATED("aggregated"), DETAIL("detail"), CURRENT("current"), ALL("all"), RAIN("rain"), STATS("stats"), UNDEFINED("undefined");

  private final String name;

  DataType(String name) {
    this.name = name;
  }

  public static DataType fromString(String name) {
    for (DataType type : DataType.values()) {
      if (type.name.equals(name)) {
        return type;
      }
    }
    return UNDEFINED;
  }

  @Override
  public String toString() {
    return name;
  }
}

File: src/main/java/org/voegtle/weatherstation/server/request/DataType.kt
package org.voegtle.weatherstation.server.request

enum class DataType private constructor(private val id: String) {
  AGGREGATED("aggregated"), DETAIL("detail"), CURRENT("current"), ALL("all"), RAIN("rain"), STATS("stats"), UNDEFINED("undefined");

  override fun toString(): String {
    return id
  }

  companion object {

    fun fromString(id: String): DataType {
      return DataType.values().firstOrNull { it.id == id }
          ?: UNDEFINED
    }
  }
}
